script/gen_requirements_all.py

```diff
@@ -157,7 +157,7 @@
 
 # protobuf must be in package constraints for the wheel
 # builder to build binary wheels
-protobuf==4.25.4
+protobuf==5.28.0
```

Comment:
```suggestion
protobuf==5.28.2
```

Reply:
FYI if running 5.28.2 it will emit 3 `UserWarnings` when loading the ring integration due to [this protobuf issue](https://github.com/protocolbuffers/protobuf/issues/18096) that has been [fixed in main](https://github.com/protocolbuffers/protobuf/pull/18406) but not yet backported to 5.28.
I compiled against 5.28.0 and it should be fine to have runtime>compiler but not the other way round. It seems `UserWarnings` are directed to the logger if I understand correctly which could cause user noise?